geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -223,6 +223,16 @@ impl IntersectionMatrix {
             && self.0[CoordPos::Outside][CoordPos::OnBoundary] == Dimensions::Empty
     }
 
+    /// Tests whether this matrix matches `[T*F**FFF*]`.
+    ///
+    /// returns `true` if the first geometry is *topologically* equal to the second.
+    pub fn is_equal_topo(&self) -> bool {
+        self.0[CoordPos::Inside][CoordPos::Inside] != Dimensions::Empty
+            && self.0[CoordPos::Inside][CoordPos::Outside] == Dimensions::Empty
+            && self.0[CoordPos::Outside][CoordPos::Inside] == Dimensions::Empty
+            && self.0[CoordPos::Outside][CoordPos::OnBoundary] == Dimensions::Empty
```

Comment:
I think we need one more condition: 

`&& self.0[CoordPos::OnBoundary][CoordPos::Outside] == Dimensions::Empty`

Reply:
Damn, thought I had them all. I looked at the other is_ methods for inspiration, but evidently got a bit confused. Which matrix entry does it correspond to?

Reply:
TBH, I firstly just intuited that any checks for an "is equal" relationship should be symmetrical - and noticed they weren't.

Then I noticed there are 5 constraints in the de-9im spec string, but you only had 4 comparisons and deduced which non-symmetrical check was missing.

But to try to answer your question directly, I think it's like this:

## Spec for "is topo equal": `[T*F**FFF*]`

|| Inside_a | OnBoundary_a | Outside_a |
| ------------- | ------------- |------ |------ |
| Inside_b |T | * | F|
| OnBounary_b | *| *| F (missing)|
| Outside_b | F | F| *|

## Corresponding Indexes into de-9im string

|| Inside_a | OnBoundary_a | Outside_a |
| ------------- | ------------- |------ |------ |
| Inside_b |0 | 1 | 2|
| OnBounary_b | 3| 4| 5 (missing)|
| Outside_b | 6 | 7| 8|

So we were missing a check for the "idx 5th" (the 6th element)

```
[T*F**FFF*]
[     ^   ]
```

Reply:
Amazing, thank you!